common/workunit/workunit.cpp

```diff
@@ -13182,13 +13182,51 @@ static void setResources(StringBuffer &jobYaml, const IConstWorkUnit *workunit,
     if (cpuLimit)
         jobYaml.replaceString("#limit-cpu", s.clear().appendf("cpu: \"%um\"", cpuLimit));
 }
-void runK8sJob(const char *name, const char *wuid)
+
+
+void deleteK8sJob(const char *componentName, const char *job)
+{
+    VStringBuffer jobname("%s-%s", componentName, job);
+    jobname.toLowerCase();
+    VStringBuffer deleteJob("kubectl delete job/%s", jobname.str());
+    StringBuffer output, error;
+    bool ret = runExternalCommand(output, error, deleteJob.str(), nullptr);
+    DBGLOG("kubectl delete output: %s", output.str());
+    if (error.length())
+        DBGLOG("kubectl delete error: %s", error.str());
+    if (ret)
+        throw makeStringException(0, "Failed to run kubectl delete");
+}
+
+void waitK8sJob(const char *componentName, const char *job, const char *condition)
 {
-    VStringBuffer jobname("%s-%s", name, wuid);
+    VStringBuffer jobname("%s-%s", componentName, job);
     jobname.toLowerCase();
-    VStringBuffer args("--workunit=%s", wuid);
+
+    if (isEmptyString(condition))
+        condition = "condition=complete";
+
+    // MORE - blocks indefinitely here if you request too many resources
+    VStringBuffer waitJob("kubectl wait --for=%s --timeout=10h job/%s", condition, jobname.str());  // MORE - make timeout configurable
+    StringBuffer output, error;
+    bool ret = runExternalCommand(output, error, waitJob.str(), nullptr);
+    DBGLOG("kubectl wait output: %s", output.str());
+    if (error.length())
+        DBGLOG("kubectl wait error: %s", error.str());
+    if (ret)
+        throw makeStringException(0, "Failed to run kubectl wait");
+}
+
+void launchK8sJob(const char *componentName, const char *wuid, const char *job, const std::list<std::pair<std::string, std::string>> &extraParams)
+{
+    VStringBuffer jobname("%s-%s", componentName, job);
+    jobname.toLowerCase();
+    VStringBuffer args("\"--workunit=%s\"", wuid);
+    for (const auto &p: extraParams)
+        args.append(',').newline().append("\"--").append(p.first.c_str()).append('=').append(p.second.c_str()).append("\"");
+    VStringBuffer jobSpecFilename("/etc/config/%s.yaml", componentName); // default to /etc/config/jobspec.yaml
```

Comment:
Does it default to that?

Reply:
no, comment is outdated, should be removed.